src/goose.rs

```diff
@@ -1221,14 +1267,14 @@ impl GooseClient {
     ///         }
     ///     }
     /// ````
-    pub fn set_failure(&mut self, method: &GooseMethod, path: &str) {
-        let request_name = self.get_request_name(path);
-        let mut update_request = GooseRawRequest::new(method.clone(), &request_name);
-        update_request.success = false;
-        // This is an updaate to a previously recorded statistic.
-        update_request.update = true;
-
-        self.send_to_parent(update_request);
+    pub fn set_failure(&mut self, request: &GooseRawRequest) {
+        // Only send update if this was previously a success.
+        if request.success {
+            let mut update_request = request.clone();
+            update_request.success = false;
+            update_request.update = true;
+            self.send_to_parent(&update_request);
```

Comment:
Is it correct that this function and send_to_parent are not async? ie send_to_parent can never block?

Reply:
Correct, we're using UnboundSender which does not block:
https://docs.rs/tokio/0.2.21/tokio/sync/mpsc/struct.UnboundedSender.html